docs/source/reference/types.rst

```diff
@@ -102,6 +102,17 @@ the beginning or the end of the index specification::
    >>> numba.float32[::1, :, :]
    array(float32, 3d, F)
 
+This style of type declaration is supported within Numba :term:`JIT` compiled
+functions e.g. declaring the type of a :ref:`typed.List <feature-typed-list>`.
+::
+
+    from numba import njit, types, typed
+
+    @njit
+    def example():
+        return typed.List.empty_list(types.float64[:, ::1])
+
+
```

Comment:
Perhaps add a note here that only simple numerical types, and not compound types, are supported?